code/functions.lua

```diff
@@ -147,6 +147,12 @@ end
 
 
 function TeleportToIsland(a_Player, a_IslandInfo)
+    if not(a_Player:GetWorld():GetDimension() == dimOverworld) then
+        a_Player:MoveToWorld(SKYBLOCK, true)
+        SKYBLOCK:ScheduleTask(2, function ()
+        TeleportToIsland(a_Player, a_IslandInfo)
+        end)
+    end
```

Comment:
I pushed a change into master that hopefully fixes that better. Accoding to Matthias I need to send the respawn packet.